gems/sorbet-runtime/lib/types/non_forcing_constants.rb

```diff
@@ -52,4 +52,31 @@ def self.non_forcing_is_a?(val, klass)
 
     current_klass.===(val)
   end
+
+  class InheritanceChecked
+    T::Sig::WithoutRuntime.sig {params(classname: String).void}
+    def initialize(classname)
+      @classname = T.let(classname.delete_prefix("::"), String)
+    end
+
+    T::Sig::WithoutRuntime.sig {returns(String)}
+    def get
+      @classname
+    end
+  end
+
+  # Note that this method is mostly used for static enforcement. At runtime, it does no inheritance checking.
+  T::Sig::WithoutRuntime.sig {params(classname: String, base: T::Class[T.anything]).returns(InheritanceChecked)}
```

Comment:
It would be very cool if `InheritanceChecked` carried around the class from which inheritance was checked, to reduce the risk of programming mistakes.  I think we'd probably have to have a non-genericized version too, though.

Reply:
Ah interesting.